### PR TITLE
Add automated test scripts to run in pipeline for the Galasa CLI - Issue 1413

### PR DESCRIPTION
**Pull request by jadecarino**
- Two test scripts written that can be ran locally but are designed to run as automated tests in the CLI main build pipeline to test the build binaries of galasactl.

Signed-off-by: Jade Carino <carino_jade@yahoo.co.uk>

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: test-galasactl-local.sh [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+    BUNDLE=$1
+    JAVA_CLASS=$2
+    OBR_GROUP_ID=$3
+    OBR_ARTIFACT_ID=$4
+    OBR_VERSION=$5
+    LOG_FILE=$6
+
+    # Could get this bootjar from https://development.galasa.dev/main/maven-repo/obr/dev/galasa/galasa-boot/0.26.0/
+    export BOOT_JAR_VERSION="0.26.0"
```
Comment (techcobweb) [inline test-galasactl-local.sh:187]: Don't think we need this variable ?

File: test-galasactl-local.sh [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+    export GALASA_VERSION="0.26.0"
+
+    export M2_PATH=$(cd ~/.m2 ; pwd)
+    export BOOT_JAR_PATH=~/.galasa/lib/${GALASA_VERSION}/galasa-boot-${BOOT_JAR_VERSION}.jar
```
Comment (techcobweb) [inline test-galasactl-local.sh:192]: variable not needed ?

File: test-galasactl-local.sh [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    if [[ "${rc}" != "0" ]]; then
+        error " Failed to build the generated source code which galasactl created."
+        exit 1
+    fi
+    success "OK"
+}
+
+#--------------------------------------------------------------------------
+# Run test using the galasactl locally in a JVM
+function submit_local_test {
```
Comment (techcobweb) [inline test-galasactl-local.sh:173]: I wonder if, now we've got the galasactl method of calling a test, whether we need to test this java -jar method ?

**Review by techcobweb — APPROVED** [review]
Some changes suggested, but I think it's looking good.

**Comment by techcobweb** [comment]
Should we try to make it invoke a test in the production ecosystem also ? to make sure it can launch a remote test and get the result ?

File: test-galasactl-local.sh (hunk 2 above)
Comment (jadecarino) [reply test-galasactl-local.sh:192]: Removed

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: test-galasactl-local.sh (hunk 1 above)
Comment (techcobweb) [reply test-galasactl-local.sh]: It's still there ?

File: test-galasactl-local.sh (hunk 2 above)
Comment (techcobweb) [reply test-galasactl-local.sh]: It's still there ?

File: test-galasactl-local.sh [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+#--------------------------------------------------------------------------
+# Invoke the galasactl command to create a project.
+function generate_sample_code {
+    h2 "Invoke the tool to create a sample project."
+
+    cd ${BASEDIR}/temp
+
+    export PACKAGE_NAME="dev.galasa.example.banking"
+    ${BASEDIR}/bin/${binary} project create --package ${PACKAGE_NAME} --features payee --obr --${buildTool} --force --development --log -
```
Comment (techcobweb) [inline test-galasactl-local.sh:168]: Should we be using the --maven or --gradle flags, based on the mode the script is running in ?

File: test-galasactl-local.sh [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+    rm -fr ~/.m2/repository/dev/galasa/example
+}
+
+# Initialise Galasa home ...
+galasa_home_init
+
+# Generate sample project ...
+generate_sample_code
+
+if [[ "${buildTool}" == "maven" ]]; then
```
Comment (techcobweb) [inline test-galasactl-local.sh:259]: These two branches do the same thing ?

**Comment by galasa-team** [comment]
Build successful

**Review by techcobweb — APPROVED** [review]
Some comments on the local script. But approved assuming you take a look at them.

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**